packages/contracts/test/voting/multisig.ts

```diff
@@ -213,118 +237,193 @@ describe('Multisig', function () {
+          .withArgs(id, signers[1].address, dummyMetadata, []);
+      });
+    });
 
-      const proposal = await multisig.getProposal(id);
-      expect(proposal.executed).to.equal(false);
-      expect(proposal.parameters.snapshotBlock).to.equal(block.number - 1);
-      expect(proposal.parameters.minApprovals).to.equal(minApprovals);
+    context('`onlyListed` is set to `true`:', async () => {
+      beforeEach(async () => {
+        multisigSettings.onlyListed = true;
 
-      expect(proposal.tally.approvals).to.equal(0);
-      expect(proposal.tally.addresslistLength).to.equal(1);
+        await multisig.initialize(
+          dao.address,
+          addresslist(1), // signers[0] is listed
+          multisigSettings
+        );
+      });
 
-      expect(await multisig.canApprove(id, signers[0].address)).to.equal(true);
-      expect(await multisig.canApprove(id, signers[1].address)).to.equal(false);
-      expect(await multisig.canApprove(1, signers[0].address)).to.equal(false);
+      it('reverts if the user is not on the list and only listed accounts can create proposals', async () => {
+        await expect(
+          multisig
+            .connect(signers[1]) // not listed
+            .createProposal(dummyMetadata, [], false, false)
+        ).to.be.revertedWith(
+          customError('ProposalCreationForbidden', signers[1].address)
+        );
+
+        await expect(
+          multisig
+            .connect(signers[0])
+            .createProposal(dummyMetadata, [], false, false)
+        ).to.not.be.reverted;
+      });
 
-      expect(proposal.actions.length).to.equal(1);
-      expect(proposal.actions[0].to).to.equal(dummyActions[0].to);
-      expect(proposal.actions[0].value).to.equal(dummyActions[0].value);
-      expect(proposal.actions[0].data).to.equal(dummyActions[0].data);
-    });
+      it('creates a proposal successfully and does not approve if not specified', async () => {
+        await expect(multisig.createProposal(dummyMetadata, [], false, false))
+          .to.emit(multisig, PROPOSAL_EVENTS.PROPOSAL_CREATED)
+          .withArgs(id, signers[0].address, dummyMetadata, []);
+
+        const block = await ethers.provider.getBlock('latest');
+
+        const proposal = await multisig.getProposal(id);
+        expect(proposal.executed).to.equal(false);
+        expect(proposal.parameters.snapshotBlock).to.equal(block.number - 1);
+        expect(proposal.parameters.minApprovals).to.equal(
+          multisigSettings.minApprovals
+        );
+        expect(proposal.tally.approvals).to.equal(0);
+        expect(proposal.tally.addresslistLength).to.equal(1);
+        expect(proposal.actions.length).to.equal(0);
+
+        expect(await multisig.canApprove(id, signers[0].address)).to.equal(
+          true
+        );
+        expect(await multisig.canApprove(id, signers[1].address)).to.equal(
+          false
+        );
+        expect(await multisig.canApprove(1, signers[0].address)).to.equal(
```

Comment:
I dON't ge what this tests. `1` would not be even created. Pointless. The test name suggests that it tests something else

Reply:
Fixed in 0f663c4521f126ff0eede25de03766ce88a69293